### PR TITLE
CI: remove obsolete jobs, and move macOS and conda Azure jobs to Meson and GHA

### PR DESCRIPTION
**Pull request by rgommers**
See commit messages of individual commits for more context. Should be straightforward (assuming CI passes). Some of the code coverage measurements is an easy delete, because that's a ton of boilerplate that does pretty much nothing. We should have code coverage in one job only, and keep all other jobs readable.

Addresses a few tasks from gh-24410.

### CODE REVIEW

**Comment by rgommers** [comment]
Hmm, the macOS OpenBLAS installs on Azure were already broken, and despite the title were disabled with `BLAS=None` etc. env vars. That's a little useless. It seems like `openblas.pc`/`openblas64.pc` isn't copied to `/usr/local/lib/`. I think I'll keep it at `-Dallow-noblas` for now, and we can fix the problem later once we have OpenBLAS wheels.

**Comment by rgommers** [comment]
Debugging Azure, so much fun. I think this has to do with picking up the local repo's `numpy` rather than the installed version - but it could be related to version git has stuff too:
```
ImportError while loading conftest '/Users/runner/work/1/s/numpy/conftest.py'.
numpy/__init__.py:101: in <module>
    from . import version
E   ImportError: cannot import name 'version' from partially initialized module 'numpy' (most likely due to a circular import) (/Users/runner/work/1/s/numpy/__init__.py)
```

**Comment by rgommers** [comment]
I can't believe that those macOS jobs are still failing. I'm inclined to throw them out and implement new macOS x86-64 jobs on GitHub Actions, that's going to be faster and more future-proof. Any objections?

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,12 @@
 [build-system]
 build-backend = "mesonpy"
-backend-path = ['./vendored-meson/meson-python']
+#backend-path = ['./vendored-meson/meson-python']
 requires = [
     "Cython>=3.0",
     # All dependencies of the vendored meson-python (except for meson, because
     # we've got that vendored too - that's the point of this exercise).
+    'meson-python',
+    'vendored-meson/meson',
```
Comment (mattip) [inline pyproject.toml:9]: This doesn't seem to work. Does it work offline? All the builds fail for this line.

Comment (rgommers) [reply pyproject.toml:9]: accidental commit, never mind that one.

**Comment by rgommers** [comment]
Here are representative timings for the 3 removed Azure jobs:

image

The new macOS conda job on GHA takes ~23 minutes on an empty cache, and ~14 minutes when the cache is populated. So we're gaining ~3x here in total CI runtime.

**Comment by rgommers** [comment]
Okay, this is happy now. The one failure in the Emscripten job is unrelated. The removed refguide check from the Azure macOS jobs should go into the CircleCI refguide check, but I'm leaving that to a follow-up to not touch three different CI systems at once.

**Comment by charris** [comment]
Thanks Ralf.

**Pull request merged**